src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.CodeFixes/IdentifierUpgradeCodeFixer.cs

```diff
@@ -60,78 +71,80 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+            var documentRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var generator = SyntaxGenerator.GetGenerator(document);
 
-            var name = node switch
-            {
-                // The most common case is that the node is an IdentifierName
-                NameSyntax n => n,
-
-                // In some cases (when the node is a SimpleBaseType, for example) we need to get the name from a child node
-                SyntaxNode x => x.DescendantNodesAndSelf().FirstOrDefault(n => n is NameSyntax)
-            };
-
-            // If we can't find a name, bail out
-            if (name is null)
+            if (documentRoot is null)
             {
                 return document;
             }
 
-            // Make sure the name syntax node includes the whole name in case it is qualified
-            if (name.Parent is NameSyntax)
-            {
-                while (name.Parent is NameSyntax)
-                {
-                    name = name.Parent;
-
-                    if (name is null)
-                    {
-                        return document;
-                    }
-                }
-            }
+            // Split the new idenfitier into namespace and name components
+            var namespaceDelimiterIndex = newIdentifier.LastIndexOf('.');
+            var (qualifier, name) = namespaceDelimiterIndex >= 0
+                ? (newIdentifier.Substring(0, namespaceDelimiterIndex), newIdentifier.Substring(namespaceDelimiterIndex + 1))
+                : ((string?)null, newIdentifier);
 
             // Create new identifier
-            SyntaxNode updatedName = SyntaxFactory.ParseName(newIdentifier)
-                .WithTriviaFrom(name)
-                .WithAdditionalAnnotations(Simplifier.Annotation);
-
-            // In some cases (accessing a static member), the name may be part of a member access expression chain
-            // instead of a qualified name. If that's the case, update the name and updatedName accordingly.
-            if (name.Parent is MemberAccessExpressionSyntax m && m.Name.ToString().Equals(name.ToString(), StringComparison.Ordinal))
-            {
-                name = name.Parent;
-
-                if (name is null)
-                {
-                    return document;
-                }
-
-                updatedName = SyntaxFactory.ParseExpression(newIdentifier)
-                    .WithTriviaFrom(name)
-                    .WithAdditionalAnnotations(Simplifier.Annotation);
-            }
-
-            // Work on the document root instead of updating nodes with the editor directly so that
-            // additional changes can be made later (adding using statement) if necessary.
-            if (editor.OriginalRoot is not CompilationUnitSyntax documentRoot)
+            // Note that the simplifier does not recognize using statements within namespaces, so this
```

Comment:
is this a known issue being tracked? shouldn't it do this?

Reply:
I'm not sure. I'll look for an issue. It's possible it's by design because recognizing using statements within a namespace would mean that the simplifier would have to consider the scope of those namespaces which is trickier than just recognizing an import that's global for the whole file.

Reply:
Makes sense. I'll see if I can find something to link to.